pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
VersionedPrinter is irrelevant for HumanReadablePrinters - can we unwrap
it, or will we need to change the factory to let us get a human readable
printer directly (so we can wrap it ourselves)?

On Fri, Jul 31, 2015 at 9:40 AM, Marcin Wielgus <notifications@github.com>
wrote:

> In pkg/kubectl/resource_printer.go
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11440#discussion_r35973706>
> :
>
> >  }
> >
> >  // ResourcePrinterFunc is a function that can print objects
> > -type ResourcePrinterFunc func(runtime.Object, io.Writer) error
> > +type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
>
> @smarterclayton <https://github.com/smarterclayton>
>
> Adding extra wrapper for HumanReadablePrinter will not work because there
> is already an extra wrapper on top of it - VersionedPrinter.
> The wrapper is applied here:
>
> https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/util/factory.go#L397
>
> https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/kubectl/cmd/get.go#L125
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleCloudPlatform/kubernetes/pull/11440/files#r35973706>
> .
>



-- 
Clayton Coleman | Lead Engineer, OpenShift